services/web3id-issuer/src/main.rs

```diff
@@ -268,13 +300,43 @@ async fn issue_credential(
         amount: Amount::zero(),
     };
 
+    let holder_id = request
+        .credential_subject
+        .id
+        .ty
+        .extract_public_key()
+        .ok_or(Error::InvalidId)?
+        .into();
+    if request.credential_subject.id.network != state.network {
+        return Err(Error::InvalidId);
+    }
+    let valid_from = Timestamp::from_timestamp_millis(
+        u64::try_from(request.valid_from.timestamp_millis())
+            .map_err(|_| Error::InvalidTimeRange)?,
+    );
+    let valid_until = request
+        .valid_until
+        .map(|v| {
+            u64::try_from(v.timestamp_millis())
+                .map_err(|_| Error::InvalidTimeRange)
+                .map(Timestamp::from_timestamp_millis)
+        })
+        .transpose()?;
+    let cred_info = CredentialInfo {
+        holder_id,
+        holder_revocable: request.holder_revocable,
+        valid_from,
+        valid_until,
+        metadata_url: request.metadata_url.clone(),
+    };
+
     let tx_hash = state
         .client
-        .register_credential(&*state.issuer, &metadata, &request.credential, &[])
+        .register_credential(&*state.issuer, &metadata, &cred_info, &[])
```

Comment:
I think it would make sense to validate the credential attributes against the credential schema before this point. This would ensure that the issued credential actually conforms to the credential type for this registry.

Reply:
I don't think it makes sense to add this.

This service is meant to handle the on-chain part of the issuance of credentials, to help with that.

So it should check for configuration errors, but I don't think it is valuable to prevent this mismatch. The person who is trying to issue this credentials, which is another service, should be trusted to issue the credentials it wants.

Reply:
It would add quite a bit of complexity to validate this according to the schema. Maybe if this takes off we can consider it.